dali/base/dadfs.cpp

```diff
@@ -5116,16 +5119,19 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
                 if (!orderedSubFiles[i])
                     ThrowStringException(-1, "CDistributedSuperFile: SuperFile %s: missing subfile part number %d of %d", logicalName.get(), i+1, n);
             }
+            bool superfileRestricted = root->getPropBool("@restricted", false);
             // Now try to resolve them all (file/superfile)
             for (unsigned f=0; f<n; f++)
             {
                 IPropertyTree &sub = *(orderedSubFiles[f]);
                 sub.getProp("@name",subname.clear());
                 Owned<IDistributedFile> subfile;
-                subfile.setown(transaction?transaction->lookupFile(subname.str(),timeout):parent->lookup(subname.str(), udesc, false, false, false, transaction, timeout));
+                subfile.setown(transaction?transaction->lookupFile(subname.str(),timeout):parent->lookup(subname.str(), udesc, false, false, false, transaction, true, timeout));
                 if (!subfile.get())
                     subfile.setown(transaction?transaction->lookupSuperFile(subname.str(),timeout):parent->lookupSuperFile(subname.str(),udesc,transaction,timeout));
-
+                subfileRestricted = subfileRestricted || subfile->isRestrictedAccess();
```

Comment:
trivial: could be:
```
subfileRestricted |= subfile->isRestrictedAccess();
```

Reply:
Isn't that bitwise 'or' rather than logical 'or'?

Reply:
it is indeed, I was just testing :)